base/simdloop.jl

```diff
@@ -54,7 +67,7 @@ function compile(x)
     (isa(x, Expr) && x.head == :for) || throw(SimdError("for loop expected"))
     length(x.args) == 2 || throw(SimdError("1D for loop expected"))
```

Comment:
Yes, the check is. The error message is (and was) wrong; the length of `args` here is independent of whether the loop is 1D or not.